packages/hawtio/src/plugins/camel/routes/CamelRoutes.tsx

```diff
@@ -0,0 +1,313 @@
+    setIsConfirmDeleteOpen(!isConfirmDeleteOpen)
+  }
+
+  const dropdownItems = [
+    <DropdownItem key='action' componentID='deleteAction'>
+      <Button
+        variant='control'
+        isDisabled={!isSuspendEnabled('Stopped')}
+        isSmall={true}
+        icon={React.createElement(Remove2Icon)}
+        onClick={onDeleteClicked}
+      >
+        Delete
+      </Button>
+    </DropdownItem>,
+  ]
+  const toolbarButtons = (
+    <React.Fragment>
+      <ToolbarItem>
+        <Button
+          variant='secondary'
+          isSmall={true}
+          isDisabled={!isSuspendEnabled('Stopped')}
+          icon={React.createElement(PlayIcon)}
+          onClick={onStartClicked}
+        >
+          Start
+        </Button>
+      </ToolbarItem>
+      <ToolbarItem>
+        <Button
+          variant='secondary'
+          isSmall={true}
+          isDisabled={!isSuspendEnabled('Started')}
+          icon={React.createElement(AsleepIcon)}
+          onClick={onStopClicked}
+        >
+          Stop
+        </Button>
+      </ToolbarItem>
+    </React.Fragment>
+  )
+
+  function onDeleteConfirmClicked() {
+    let deleted = 0
+    selected.forEach(async (value, key) => {
+      if (value) {
+        const route = routes.find(r => r.RouteId === key)
+        if (route && route.State === 'Stopped') {
+          try {
+            await routesService.deleteRoute(route.objectName as string)
+            deleted++
+            workspace.refreshTree()
+          } catch (error) {
+            eventService.notify({
+              type: 'danger',
+              message: `Couldn't delete the route: ${error}`,
+            })
+          }
+        }
+      }
+    })
+    if (deleted > 0) {
+      eventService.notify({
+        type: 'success',
+        message: `${deleted} routes deleted successfully`,
+      })
+    }
+    setSelected(new Map())
+    setRoutes([])
+    setIsConfirmDeleteOpen(false)
+  }
+
+  const ConfirmDeleteModal = () => (
+    <Modal
+      variant={ModalVariant.small}
+      title='Are you sure?'
+      titleIconVariant='danger'
+      isOpen={isConfirmDeleteOpen}
+      onClose={handleConfirmDeleteToggle}
+      actions={[
+        <Button key='delete' variant='danger' onClick={onDeleteConfirmClicked}>
+          Delete
+        </Button>,
+        <Button key='cancel' variant='link' onClick={handleConfirmDeleteToggle}>
+          Cancel
+        </Button>,
+      ]}
+    >
+      <p>You are about to delete selected camel routes.</p>
+      <p>This operation cannot be undone so please be careful.</p>
+    </Modal>
+  )
+  return (
+    <Page>
+      {routes.length === 0 && (
+        <Card>
+          <CardBody>
+            <Text component='p'>
+              <InfoCircleIcon /> This context does not have any routes.
+            </Text>
+          </CardBody>
+        </Card>
+      )}
+      {routes.length > 0 && (
+        <TableComposable aria-label='Simple table' variant={'compact'}>
```

Comment:
In `InflightExchanges`, I used the react legacy table. If we are to go with `<TableComposable>` here then perhaps an issue to convert the `inflight` table to that format too?

Reply:
We should eventually migrate all tables used in Hawtio next to the new `<TableComposable>`, so I think we can leave it as is for this page. We should replace the usages of legacy table with `TableComposable` in the other places including Camel Inflight and JMX attributes etc. step by step. But it's not high priority yet.

Let's choose `TableComposable` when creating a new view from now on.

Reply:
Issue logged. https://github.com/hawtio/hawtio-next/issues/180